<launch>
  <arg name="ndt_slam_param_path" default="$(find-pkg-share ndt_slam)/config/ndt_slam.param.yaml"/>

  <arg name="rviz" default="true"/>

  <arg name="input_sensor_topic" default="/velodyne_points"/>
  <arg name="input_imu_topic" default="/imu"/>

  <node pkg="ndt_slam" exec="ndt_slam_node" name="ndt_slam_node" output="screen">
    <remap from="points_raw" to="$(var input_sensor_topic)"/>
    <remap from="imu" to="$(var input_imu_topic)"/>
    <param from="$(var ndt_slam_param_path)"/>
  </node>

  <group if="$(var rviz)">
    <node pkg="rviz2" exec="rviz2" name="rviz2" args="-d $(find-pkg-share ndt_slam)/rviz/ndt_slam.rviz"/>
  </group>
</launch>
